lib/pry/history.rb

```diff
@@ -78,8 +80,26 @@ def to_a
       @history.dup
     end
 
+    # Filter the history with the histignore options
+    # @return [Array<String>] An array containing all the lines that are not included
+    # in the histignore.
+    def filter(history)
+      history.select { |l| l unless should_ignore?(l) }
```

Comment:
IMHO <code>select</code> is clearer, it really show my intentions(selecting from history unless a conditions is met), also <code>map</code> behaves different. e.g:

```ruby
a = [1,2,3,4,5,6,7,8,9,10]
a.map { |n| n if n.odd? } #=> [1, nil, 3, nil, 5, nil, 7, nil, 9, nil]

a.select {|n| n if n.odd?} #=> [1, 3, 5, 7, 9]
```